### ISSUE TITLE
 RuntimeWarning: Unknown node type <MatchMapping>

### ISSUE THREAD

**Issue by edreamleo** [python-rope/rope#522]
**Describe the bug**

Running all tests with `pytest -v -W ignore::DeprecationWarning` yields:
```console
ropetest/refactor/extracttest.py::ExtractMethodTest::test_extract_method_containing_structural_pattern_match_3
  C:\Repos\rope\rope\refactor\patchedast.py:93: RuntimeWarning: Unknown node type <MatchMapping>; please report!
```

**To Reproduce**

- git clone https://github.com/python-rope/rope.git
- pip install -e .[dev]
- pytest -v -W ignore::DeprecationWarning

**Other information**

 - Python version: Python 3.10.5
   (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
- pytest version: pytest 7.1.3
 - Rope version: 5f8bd4a788b5b
 - Text editor/IDE and version (not used in test): Leo 6.7.2-devel

**Comment by lieryan** [comment]
Thanks for reporting this issue. This should be fixed now in master and should be included for the upcoming 1.5.0 release.

**Comment by edreamleo** [comment]
@lieryan You're welcome :-)
